### PR TITLE
Set stmt use new analyzer

### PR DESCRIPTION
**Pull request by kangkaisen**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [x] refactor
- [ ] others

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 44 / 71 (61.97%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/analysis/SetType.java|0|6|00.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|0|7|00.00%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/SetNamesVar.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/load/Load.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/analysis/SetPassVar.java|7|11|63.64%|
|:large_blue_circle:|com/starrocks/analysis/SetStmt.java|3|4|75.00%|
|:large_blue_circle:|com/starrocks/analysis/SetVar.java|15|19|78.95%|
|:large_blue_circle:|com/starrocks/planner/LoadScanNode.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/AdminStmtAnalyzer.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/analysis/SetUserPropertyVar.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/analysis/SetUserPropertyStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/analysis/Expr.java|6|6|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/ScalarOperatorRewriter.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/mysql/MysqlPassword.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/cluster/ClusterNamespace.java|2|2|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 44 / 72 (61.11%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/analysis/SetType.java|0|6|00.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|0|7|00.00%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/SetNamesVar.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/load/Load.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/analysis/SetStmt.java|3|5|60.00%|
|:large_blue_circle:|com/starrocks/analysis/SetPassVar.java|7|11|63.64%|
|:large_blue_circle:|com/starrocks/analysis/SetVar.java|15|19|78.95%|
|:large_blue_circle:|com/starrocks/planner/LoadScanNode.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/AdminStmtAnalyzer.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/analysis/SetUserPropertyVar.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/analysis/SetUserPropertyStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/analysis/Expr.java|6|6|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/ScalarOperatorRewriter.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/mysql/MysqlPassword.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/cluster/ClusterNamespace.java|2|2|100.00%|

**Review by murphyatwork — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/ScalarOperatorRewriter.java [hunk 1]
```diff
@@ -23,7 +23,7 @@ public class ScalarOperatorRewriter {
             new ImplicitCastRule()
     );
 
-    public static final List<ScalarOperatorRewriteRule> LOAD_REWRITE_RULES = Lists.newArrayList(
+    public static final List<ScalarOperatorRewriteRule> CastAndFold_RULES = Lists.newArrayList(
```
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/ScalarOperatorRewriter.java:26]:
```suggestion
    public static final List<ScalarOperatorRewriteRule> CAST_AND_FOLD_RULES = Lists.newArrayList(
```

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 44 / 72 (61.11%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/analysis/SetType.java|0|6|00.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|0|7|00.00%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/SetNamesVar.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/load/Load.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/analysis/SetStmt.java|3|5|60.00%|
|:large_blue_circle:|com/starrocks/analysis/SetPassVar.java|7|11|63.64%|
|:large_blue_circle:|com/starrocks/analysis/SetVar.java|15|19|78.95%|
|:large_blue_circle:|com/starrocks/planner/LoadScanNode.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/AdminStmtAnalyzer.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/analysis/SetUserPropertyVar.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/analysis/SetUserPropertyStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/analysis/Expr.java|6|6|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/ScalarOperatorRewriter.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/mysql/MysqlPassword.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/cluster/ClusterNamespace.java|2|2|100.00%|

**Review by Seaven — APPROVED** [review]

**Review by murphyatwork — APPROVED** [review]

**Pull request merged**